src/Model/DecTssQcModel.h

```diff
@@ -0,0 +1,138 @@
+	double *		rhs_; 			/** constraint rhs of each constraint */
+	int *			sense_; 		/** constraint sense of each constraint */
+	int **        linind_; 			/** indices for the linear part */
+	double **      linval_; 		/** nonzero coefficient of the linear part */
+	int **      	quadrow_;  		/** indices for the quadratic part */
+	int **      	quadcol_;  		/** indices for the quadratic part */
+	double **     quadval_; 		/** nonzero coefficient of the quadratic part */ 
+};
+
+/**
+ * A decomposable two-stage stochastic problem with quadratic constraints. 
+ */
+class DecTssQcModel: public DecTssModel {
+
+public:
+
+	/** default constructor */
+	DecTssQcModel();
+
+	/** copy constructor */
+	DecTssQcModel(const DecTssQcModel & rhs);
+
+	/** copy constructor */
+	DecTssQcModel(const DecTssModel & rhs);
+
+	/** default destructor */
+	virtual ~DecTssQcModel();
+
+public:
+
+	bool isQuadratic() {return true;}
+
+	/** construct a map that maps variable names to their indices */
+	bool mapVarnameIndex(map<string, int> &map_varName_index, const char * corefilename);
+	
+	/** read quadratic data file */
+	DSP_RTN_CODE readQuad(const char * smps, const char * filename);
+
+	/** set dimensions for quadratic constraints */
+    void setQcDimensions(){QcRowData_.resize(nscen_);};
+	DSP_RTN_CODE setQcDimensions(int * nqrows);
+	DSP_RTN_CODE setQcDimensions(int s, int nqrows);
+
+	/** set file name */
+	void setFileName(const char * smps) {filename_ = smps;} 
+
+	/** add quadratic constraints to the second stage problem */
+	DSP_RTN_CODE loadQuadraticRows (
+			const int           s,     		/**< scenario index */
+		const int 			nqrows,
+        const int *         linnzcnt,  	/**< number of nonzero coefficients in the linear part of each constraint  */
+        const int *        	quadnzcnt,  /**< number of nonzero coefficients in the quadratic part of each constraint  */
+		const double *		rhs, 		/**< constraint rhs of each constraint */
+		const int *			sense, 		/**< constraint sense of each constraint */
+		const int *         linstart,  	/**< number of nonzero coefficients in the linear part of each constraint  */
+		const int *         linind, 	/**< indices for the linear part */
+		const double *      linval, 	/**< nonzero coefficient of the linear part */
+		const int *        	quadstart,  /**< number of nonzero coefficients in the quadratic part of each constraint  */
+		const int *       	quadrow,  	/**< indices for the quadratic part */
+		const int *       	quadcol,  	/**< indices for the quadratic part */
+		const double *      quadval 	/**< nonzero coefficient of the quadratic part */ );
+	
+	/** print quadratic constraints */
+	DSP_RTN_CODE printQuadRows (const int s);
+
+	/** print quadratic constraints for given params */
+	DSP_RTN_CODE printQuadRows (QcRowDataScen *qcdata);
+
+	/** get parameters for CPX interface */
+	QcRowDataScen * getQRowsCPXParams (int s) {return &QcRowData_[s];}
```

Comment:
What does this function do? Can we avoid to define this function at `DspOsi` level, instead of being solver-specific?

Reply:
This function retrieves data for quadratic constraints. Yes, you're right. It doesn't need to be solver specific. Changed the function name to getQcRowData for clarification.